spark/v3.3/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ExtendedDistributionAndOrderingUtils.scala

```diff
@@ -57,10 +61,30 @@ object ExtendedDistributionAndOrderingUtils {
         } else {
           conf.numShufflePartitions
         }
-        // the conversion to catalyst expressions above produces SortOrder expressions
-        // for OrderedDistribution and generic expressions for ClusteredDistribution
-        // this allows RepartitionByExpression to pick either range or hash partitioning
-        RepartitionByExpression(ArraySeq.unsafeWrapArray(distribution), query, finalNumPartitions)
+
+        val tableProperties = if(table.isInstanceOf[RowLevelOperationTable])  {
+          table.asInstanceOf[RowLevelOperationTable].table.properties()
```

Comment:
In case of RowLevelTableOperation, had to get the nested table to get its properties.